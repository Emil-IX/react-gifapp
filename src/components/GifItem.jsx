import React from 'react'

export const GifItem = ({ title, url, id}) => {

    // console.log(props);

  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p> {title} </p>
    </div>
  )
}
