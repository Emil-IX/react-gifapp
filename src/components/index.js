
export * from './AddCategpry';
export * from './GifGrid';
export * from './GifItem';