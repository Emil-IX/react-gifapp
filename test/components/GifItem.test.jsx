import { render,   screen  } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('test en <GifItem />', () => { 

    const title = 'Emsy';
    const url = 'https://client-island/'

    test('hacer match con el snapshot', () => {  

    const { container } = render( <GifItem title={title} url={url} />)
        expect( container ).toMatchSnapshot();

    });


    test('debe de mostrar la imagen con el url y el alt indicado', () => { 

        render(<GifItem title={title} url={url} />)
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

         const { src, alt } = screen.getByRole('img');
         expect(src).toBe(url);
         expect(alt).toBe(title);

     });

     test('debe de mostrar el titulo', () => { 

        render(<GifItem title={title} url={url} />)

        expect(screen.getByText(title)).toBeTruthy()

      });


 });