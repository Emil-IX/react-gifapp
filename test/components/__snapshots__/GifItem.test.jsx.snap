// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en <GifItem /> hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Emsy"
      src="https://client-island/"
    />
    <p>
       
      Emsy
       
    </p>
  </div>
</div>
`;
